test/hls/hls_parser_unit.js

```diff
@@ -2492,4 +2492,176 @@ describe('HlsParser', () => {
     expect(actual.variants.length).toBe(1);
     expect(actual).toEqual(manifest);
   });
+
+  describe('Variable substitution', () => {
+    it('parse variables master playlist', async () => {
+      const master = [
+        '#EXTM3U\n',
+        '#EXT-X-DEFINE:NAME="auth",VALUE="?token=1"\n',
+        '#EXT-X-STREAM-INF:BANDWIDTH=200,CODECS="avc1,mp4a",',
+        'RESOLUTION=960x540,FRAME-RATE=60,VIDEO="vid"\n',
+        'audio.m3u8{$auth}\n',
+        '#EXT-X-MEDIA:TYPE=VIDEO,GROUP-ID="vid",URI="video.m3u8{$auth}"',
+      ].join('');
+
+      const media = [
+        '#EXTM3U\n',
+        '#EXT-X-PLAYLIST-TYPE:VOD\n',
+        '#EXT-X-MAP:URI="init.mp4"\n',
+        '#EXTINF:5,\n',
+        'segment.mp4',
+      ].join('');
+
+      fakeNetEngine
+          .setResponseText('test:/host/master.m3u8', master)
+          .setResponseText('test:/host/audio.m3u8?token=1', media)
+          .setResponseText('test:/host/video.m3u8?token=1', media)
+          .setResponseValue('test:/host/init.mp4', initSegmentData)
+          .setResponseValue('test:/host/segment.mp4', segmentData)
```

Comment:
init.mp4 and segment.mp4 seem to be duplicated in this list.

Reply:
Fixed!